src/Val.cc

```diff
@@ -3379,17 +3390,17 @@ void VectorVal::ValDescribe(ODesc* d) const
 	{
 	d->Add("[");
 
-	if ( val.vector_val->size() > 0 )
-		for ( unsigned int i = 0; i < (val.vector_val->size() - 1); ++i )
+	if ( vector_val->size() > 0 )
+		for ( unsigned int i = 0; i < (vector_val->size() - 1); ++i )
 			{
-			if ( (*val.vector_val)[i] )
-				(*val.vector_val)[i]->Describe(d);
+			if ( (*vector_val)[i] )
+				(*vector_val)[i]->Describe(d);
 			d->Add(", ");
 			}
 
-	if ( val.vector_val->size() &&
-	     (*val.vector_val)[val.vector_val->size() - 1] )
-		(*val.vector_val)[val.vector_val->size() - 1]->Describe(d);
+	if ( vector_val->size() &&
+	     (*vector_val)[vector_val->size() - 1] )
+		(*vector_val)[vector_val->size() - 1]->Describe(d);
```

Comment:
Suggestion:

```
	if ( ! vector_val->empty() )
		for ( unsigned int i = 0; i < (vector_val->size() - 1); ++i )
			{
			if ( vector_val->at(i) )
				vector_val->at(i)->Describe(d);
			d->Add(", ");
			}

	if ( ! vector_val->empty() && vector_val->back() )
		vector_val->back()->Describe(d);
```

Bonus points for getting rid of that `vector_val->size()` in the loop too.